rosbag2_compression/src/rosbag2_compression/sequential_compression_writer.cpp

```diff
@@ -66,12 +65,7 @@ void SequentialCompressionWriter::compression_thread_fn()
   // Every thread needs to have its own compression context for thread safety.
   auto compressor = compression_factory_->create_compressor(
     compression_options_.compression_format);
-
-
-  if (!compressor) {
-    throw std::runtime_error{
-            "Cannot compress message; Writer is not open!"};
-  }
+  rcpputils::check_true(compressor != nullptr, "Could not create compressor.");
```

Comment:
Update header to state that this now throws `rcpputils::InvalidStateException` instead of `std::runtime_error`?